### ISSUE TITLE
Java - Support Metavariables within @Path("{$VALUE}...")

### ISSUE THREAD

**Issue by LewisArdern** [semgrep/semgrep#5620]
**Describe the bug**
There is a use-case where customers may want to match the PATH value with the PathParam value e.g.

https://semgrep.dev/s/gND5

```
      @Produces("application/json")
      @Path("{$EXPECTEDTOMATCH}...")
      public $TYPE $FUNC(@PathParam("$EXPECTEDTOMATCH") $TYPE1 $VALUE2)
```

https://r2c-community.slack.com/archives/CK86BJ5DW/p1655975606348409

**To Reproduce**
https://semgrep.dev/s/gND5

**Expected behavior**
The following code should match https://semgrep.dev/s/gND5 based on pattern written.

**What is the priority of the bug to you?**

- [x] P2: regular bug that should get fixed

**Use case**
Community slack request

"In my case, I wanna match methods that are both 1. annotated with @Path containing a parameter, and 2. reusing this path parameter name in the methods parameters."

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-1561/java-support-metavariables-within-pathdollarvalue. _Note_: this link is for r2c use only and is not accessible publicly.

**Comment by mjambon** [comment]
I don't think we can do this yet. A solution I could imagine is to first capture the strings to analyze as metavariables (`$PATH`, `$EXPECTEDTOMATCH`):
```
@Path("$PATH")
public $TYPE $FUNC(@PathParam("$EXPECTEDTOMATCH") $TYPE1 $VALUE2)
```
... and then check for the inclusion of one value into another. That last part isn't something we support at the moment. See https://github.com/returntocorp/semgrep/issues/3825

**Comment by Pamplemousse** [comment]
@LewisArdern, this does not seem specific to Java annotations.

I had expected the rule

```yaml
id: untitled_rule
message: Semgrep found a match
languages: [java]
severity: WARNING
pattern: |
  "/{$VAR}/"
```
to match the content of the parentheses, and hold `foo` in the metavariable `$VAR` from

```java
System.out.println("/{foo}/")
```

But it is not the case: https://semgrep.dev/s/9nx8 .
